nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/kafka/KafkaLogTailer.java

```diff
@@ -308,6 +309,41 @@ public void reset(LogPartition partition) {
         seek(new LogOffsetImpl(partition, beginningOffsets.get(topicPartition)));
     }
 
+    @Override
+    public void seek(long timestamp) {
+        Map<TopicPartition, Long> timestampsToSearch = new HashMap<>();
+        topicPartitions.forEach(tp -> timestampsToSearch.put(tp, timestamp));
+        seek(timestampsToSearch);
+    }
+
+    @Override
+    public void seek(LogPartition partition, long timestamp) {
+        TopicPartition topicPartition = new TopicPartition(prefix + partition.name(), partition.partition());
+        seek(Collections.singletonMap(topicPartition, timestamp));
+    }
+
+    /**
+     * Set the current position of the TopicPartition to the earliest offset
+     * whose timestamp is greater than or equal to the given timestamp
+     * @param timestampsToSearch
+     */
+    protected void seek(Map<TopicPartition, Long> timestampsToSearch) {
+        Map<TopicPartition, OffsetAndTimestamp> offsetsForTimes = consumer.offsetsForTimes(timestampsToSearch);
+
+        for (Map.Entry<TopicPartition, OffsetAndTimestamp> entry:offsetsForTimes.entrySet()) {
+            if (entry.getValue() != null) {
```

Comment:
You may want to do `if (entry.getValue() == null) { continue; }` (early exit) to avoid indenting the whole block.